shared/js/background/chrome-events.es6.js

```diff
@@ -38,6 +45,118 @@ chrome.runtime.onInstalled.addListener(function (details) {
+        settings.ready().then(onReady)
+    }
+}
+
+let onBeforeNavigateTimeStamp = null
+chrome.webNavigation.onBeforeNavigate.addListener(details => {
+    onBeforeNavigateTimeStamp = details.timeStamp
+})
+
+chrome.webNavigation.onCommitted.addListener(details => {
+    getOnboardingParams((err, { showWelcomeBanner, showCounterMessaging }) => {
+        if (err) {
+            console.error(chrome.runtime.lastError)
+        }
+
+        // We show the welcome banner and counter messaging only once
+        if (showWelcomeBanner || showCounterMessaging) {
+            const isAddressBarQuery = details.transitionQualifiers.includes('from_address_bar')
+
+            if (showWelcomeBanner) {
+                settings.removeSetting('showWelcomeBanner')
+            }
+            if (isAddressBarQuery && showCounterMessaging) {
+                settings.removeSetting('showCounterMessaging')
+            }
+
+            if (onBeforeNavigateTimeStamp < details.timeStamp) {
+                chrome.tabs.executeScript(details.tabId, {
+                    code: createOnboardingCode({
+                        isAddressBarQuery,
+                        showWelcomeBanner,
+                        showCounterMessaging,
+                        browser,
+                        extensionId: chrome.runtime.id
+                    }),
+                    runAt: 'document_end'
+                }, (response) => {
+                    if (chrome.runtime.lastError) {
+                        console.error(chrome.runtime.lastError)
+                    }
+                })
+            }
+        }
+    })
+}, {
+    url: [
+        {
+            originAndPathMatches: '^https:\\/\\/(?:[^/]*\\.)?duckduckgo\\.com/$',
+            queryContains: '?q='
+        },
+        {
+            originAndPathMatches: '^https:\\/\\/(?:[^/]*\\.)?duckduckgo\\.com/$',
+            queryContains: '&q='
+        }
+    ]
+})
+
+/**
+ * Health checks + `showCounterMessaging` mutation
+ * (Chrome only)
+ */
+if (browser === 'chrome') {
+    chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
+        if (request === 'healthCheckRequest') {
+            sendResponse(true)
+        } else if (request === 'rescheduleCounterMessagingRequest') {
+            const update = () => {
+                settings.updateSetting('rescheduleCounterMessagingOnStart', true)
+                sendResponse(true)
+            }
+            if (settings.isReady()) {
+                update()
+            } else {
+                settings.ready().then(update)
+            }
+        }
+    })
+
+    chrome.runtime.onStartup.addListener(() => {
+        const init = () => {
+            if (settings.getSetting('rescheduleCounterMessagingOnStart')) {
+                settings.removeSetting('rescheduleCounterMessagingOnStart')
+                settings.updateSetting('showCounterMessaging', true)
+            }
+        }
+        if (settings.isReady()) {
+            init()
```

Comment:
Sorry missed this one too, this could use a conditional `await` instead here

Reply:
this has been refactored to use async/await